### PR TITLE
Template for a "feature" issue

### PR DESCRIPTION
**Pull request by LSDancer**
**Fixes**
Added a template for a "feature" issue

### CODE REVIEW

**Comment by LSDancer** [comment]
Template for a "feature" issue

File: .github/ISSUE_TEMPLATE/feature.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+TODO: provide links to documentation which explains the functionality in details. Try to link the relevant information, not to replicate it here.
+
+
+
+**Development Tasks**
+
+_Shall be filled by developers._
+
+TODO: List all technical tasks which are required for feature implementation. Also other issues (bugs, documentation tasks, etc) are possible here.
```
Comment (pjanck) [inline .github/ISSUE_TEMPLATE/feature.md:29]: I oppose editing open issue descriptions. How else am I to fill in this section?

File: .github/ISSUE_TEMPLATE/feature.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+name: "\U00002611 Feature"
+about: Feature: collection of smaller tasks with a clear value for the end user
+title: "[MEMO]"
```
Comment (pjanck) [inline .github/ISSUE_TEMPLATE/feature.md]: Why does the title have `[MEMO]`?

**Review by pjanck — COMMENTED** [review]
A few clarifications are needed.

**Comment by pjanck** [comment]
Also: how is this different from the `TASK` template?

File: .github/ISSUE_TEMPLATE/feature.md (hunk 2 above)
Comment (LSDancer) [reply .github/ISSUE_TEMPLATE/feature.md]: fixed, should be FEATURE of course

File: .github/ISSUE_TEMPLATE/feature.md (hunk 1 above)
Comment (LSDancer) [reply .github/ISSUE_TEMPLATE/feature.md:29]: I dont have better ideas here... Developers shall identify and fill out the list of technical tasks, not the person who creates the new "Feature". I would suggest to start in this mode, and i am happy to discuss alternatives

**Comment by LSDancer** [comment]
Differences from TASK template:
 * Feature has a clear business value
 * Feature is a collection of development (and potentially other) tasks which implement it

**Comment by SergejMuhic** [comment]
I was actually thinking of removing the code guidelines "feature" and "issue" distinction. All tasks are connected to issues and therefore serve the same purpose.

File: .github/ISSUE_TEMPLATE/feature.md (hunk 1 above)
Comment (pjanck) [reply .github/ISSUE_TEMPLATE/feature.md:29]: I'd propose to specify the technical tasks in separate issues and link them to this issue.

Comment (GeorgDangl) [reply .github/ISSUE_TEMPLATE/feature.md:29]: Yeah, I think we're having the same ideas here, just worded differently😊

I think the idea of this `feature` template is something like a top-level issue, or a collection issue, which basically just links other issues related to a feature implementation. The list could look like this:

Feature: Implement ModelQuality use case

* [ ] #123 Provide backend
* [ ] #125 Create frontend
* [ ] #129 Validate frontend with users

**Comment by GeorgDangl** [comment]
I think that our original idea was to have something different from a `task`, so we came up with the idea of a `feature`. It's more broad than a single task, since it's intended to describe a, well, `feature` from the end-users point of view and also collect all tasks required for it.

That way, we'd be able to track the progress of a specific issue, like _feature foo is still waiting on finalizing the bar, then the implementation of the fizz can start_

**Comment by SergejMuhic** [comment]
> 
> 
> I was actually thinking of removing the code guidelines "feature" and "issue" distinction. All tasks are connected to issues and therefore serve the same purpose.

The feature definitely conflicts with the dev team direction where we actually want to move away from too much redundant work outside the solution. Hence, recruiting @MS-SMNS-BIM first to avoid the excel preparation. This would then move forward for the rest.

File: .github/ISSUE_TEMPLATE/feature.md (hunk 1 above)
Comment (LSDancer) [reply .github/ISSUE_TEMPLATE/feature.md:29]: this is the idea, the issues (tech tasks, bugs, maybe something else) will be linked to the feature, not defined in the feature

Comment (pjanck) [reply .github/ISSUE_TEMPLATE/feature.md:29]: I see.

> _Shall be filled by developers._

I'm not keen to change a description of an issue that wasn't authored by me. 

And looking at some issues in Portal: it seems that neither are _product owners_ the authors of the issues following this template. Therefore: 

1. Are these issues mere placeholders?
2. Are back-end developers even planned to fill in anything at anytime, or is this template more directed at the front-end developers?

**Review by pjanck — APPROVED** [review]
Looking at how it was used in another repository, it looks fine. Whatever floats your boat and makes you productive!

**Comment by pjanck** [comment]
We now use Projects for this. Closing.

**Pull request closed without merging**